### PR TITLE
SDIT-707 Create prison events from XTag events

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/config/OpenApiConfiguration.kt [hunk 1]
```diff
@@ -32,7 +32,7 @@ class OpenApiConfiguration(buildProperties: BuildProperties) {
     .info(
       Info().title("HMPPS Prisoner Events API")
         .version(version)
-        .description("Controls dequeuing and publishing NOMIS events to the Prisoner Events topic")
+        .description(javaClass.getResource("/swagger-description.html")!!.readText())
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/config/OpenApiConfiguration.kt:35]: moved some of the swagger documentation from the prison-offender-events repo

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/service/XtagEventsService.kt [hunk 2]
```diff
@@ -36,6 +35,11 @@ class XtagEventsService(
             oe.movementType = this.movementType
           }
       }
+
+      "PERSON_RESTRICTION-UPSERTED", "PERSON_RESTRICTION-DELETED" -> {
+        oe as PersonRestrictionOffenderEvent
+        oe.offenderIdDisplay = sqlRepository.getNomsIdFromRestriction(oe.offenderPersonRestrictionId!!).firstOrNull()
+      }
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/service/XtagEventsService.kt:42]: decorate event with offenderIdDisplay

**Review by andymarke — APPROVED** [review]
lgtm

File: src/main/resources/swagger-description.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        <li><b>publishedAt</b> the zoned timestamp when this events was published e.g 2020-12-04T10:42:43+01:00, 2020-12-04T10:42:43Z</li>
+      </ul>
+    </p>
+</div>
+
+<h3>Topic subscription</h3>
+<p>Clients are expected to use a SQS AWS queue to receive events with queue subscribed to <b>prison-events-topic</b> or <b>hmpps-domain-topic</b> or both. </p>
+<p>Clients can subscribe to one or more events. A typical subscription could be:</p>
+<pre>
+    resource "aws_sns_topic_subscription" "my_probation_subscription" {
```
[the remark this answers is not in the document]
Comment (steverendell) [reply src/main/resources/swagger-description.html:30]: ok, will fix next time

**Pull request merged**